### PR TITLE
Singleton design pattern implementation for UI server client:

### PR DESCRIPTION
**Pull request by jerome-benoit**
Partial code review + proper implementation as a learning exercice: 

+ singleton class shall never use static methods: it's redundant, the
  singleton already guarantee that all invocations will occur on the very
  same instance
+ the singleton is also a proxy class for WebSocket API, the naming
  shall reflect it

Signed-off-by: Jérôme Benoit <jerome.benoit@sap.com>

### CODE REVIEW

File: src/ui/web/src/components/charging-stations/CSConnector.vue [hunk 1]
```diff
@@ -61,9 +61,9 @@ function hideTagModal(): void {
 }
 
 function startTransaction(): void {
-  UIServer.startTransaction(props.hashId, props.connectorId, state.tag);
+  UIClient.instance.startTransaction(props.hashId, props.connectorId, state.tag);
```
Comment (HazbinFaulted) [inline src/ui/web/src/components/charging-stations/CSConnector.vue:57]: The reason i made all method static was to not have the redundancy of instance. Is it not "cleaner" the old way ?

Comment (jerome-benoit) [reply src/ui/web/src/components/charging-stations/CSConnector.vue:57]: Either you use a class with static methods only, fordid instantiation, static attributes to ensure that they will be always put in memory once, either you implement a singleton class but not both. 
Looks at your original code: 
+ indirection with setters to get the running instance safely (useless branching) in hot code path inside the class
+ indirection with setters to access the class attributes safely: the whole purpose of OOP is to do class that permits to access its attributes directly, without overhead and safely per design. 

The singleton design pattern UML classes diagram: https://en.wikipedia.org/wiki/Singleton_pattern

Comment (HazbinFaulted) [reply src/ui/web/src/components/charging-stations/CSConnector.vue:57]: I understand the problem with branching, didn't think of it.

**Comment by HazbinFaulted** [comment]
If everyone is okay with it i shall merge tomorrow by 10am (GMT+2)

**Pull request merged**